assets/src/components/with-placeholder-featured-image.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * WordPress dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+
+/**
+ * External dependencies
+ */
+import { delay } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { hasMinimumDimensions } from './';
+
+/**
+ * Gets a wrapped version of MediaPlaceholder.
+ *
+ * On uploading an image, if there is no featured image, set it to the selected image.
+ * The InitialMediaPlaceholder is usually the MediaPlaceholder component, unless it was filtered somewhere.
+ *
+ * @param {Function} initialPlaceholder The MediaPlaceholder component, passed from the filter.
+ * @return {Function} The wrapped component.
+ */
+export default ( initialPlaceholder ) => {
+	return class FeaturedImageMediaPlaceholder extends initialPlaceholder {
+		/**
+		 * Construct the class.
+		*/
+		constructor() {
+			super( ...arguments );
+			this.componentWillUnmount = this.componentWillUnmount.bind( this );
+		}
+
+		/**
+		 * The handler for before the component unmounts.
+		 *
+		 * If there's no featured image, set this image as the featured image.
+		 * This component unmounts right after uploading an image.
+		 * But this needs to be delayed, as there's an apiFetch() call that gets the id.
```

Comment:
Where is that `apiFetch()` call happening? Can we somehow avoid using `delay()` by `await`ing the result of that or something?

Reply:
Good question, I'll look at this more. The `apiFetch()` call happens when the `MediaUploader` calls [createMediaFromFile()](https://github.com/WordPress/gutenberg/blob/285875a1e2899b4999bb1396277be8a163354249/packages/editor/src/utils/media-upload/media-upload.js#L214).

Reply:
https://github.com/ampproject/amp-wp/pull/1998/commits/7a3df1afbe8c35d8170d213de78611e40af3016d prevents using `delay()` by instead filtering `BlockEdit`.